### PR TITLE
Ft-notify-jobs-running-too-often-155965579

### PR DESCRIPTION
**Pull request by Kalela**
### What does this PR do?
Creates functionality for notifying a user if their cron job runs too often. Tests included.
### How should this be manually tested?
- Run tests using python manage.py test
### OR
- Run the server and open healthchecks.
- Create a check and set the timeout and grace period.
- Run the check in your browser to ping it.
- Run the check again before the timeout ends.
- Status should change to Often to notify the user.
### What are the relevant pivotal tracker stories?
User should be notified when a job is run too often
ID: #155965579

### CODE REVIEW

**Review by lakivisi-zz — APPROVED** [review]

File: hc/front/tests/test_my_checks.py [hunk 1]
```diff
@@ -58,3 +58,19 @@ def test_it_shows_amber_check(self):
 
         # Mobile
         self.assertContains(r, "label-warning")
```
Comment (FelixWambiri) [inline hc/front/tests/test_my_checks.py:60]: Good tests
Good colour options for the often icon
Good and comprehensive pull request

File: hc/api/views.py [hunk 2]
```diff
@@ -118,13 +118,15 @@ def badge(request, username, signature, tag):
...
+
         if check.get_status == "up":
             status = "often"
 
         if check.get_status() == "down":
             status = "down"
             break
 
+    print(status)
+    print(tag)
```
Comment (betsybeth) [inline hc/api/views.py]: Remove the print statements

File: hc/front/tests/test_my_checks.py [hunk 3]
```diff
@@ -58,3 +58,19 @@ def test_it_shows_amber_check(self):
...
+
+    def test_it_shows_purple_check(self):
```
Comment (meshnesh) [inline hc/front/tests/test_my_checks.py:62]: nice implementation of the test for checking how it's checking

File: hc/api/views.py [hunk 4]
```diff
@@ -22,6 +22,17 @@ def ping(request, code):
         return HttpResponseBadRequest()
 
     check.n_pings = F("n_pings") + 1
+    if check.last_ping:
```
Comment (meshnesh) [inline hc/api/views.py:25]: good implementation

**Review by FelixWambiri — COMMENTED** [review]
Good on the testing
Good colour choice on the often option
Good and comprehensive pull request.

File: hc/api/migrations/0027_auto_20180328_1452.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10 on 2018-03-28 14:52
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
```
Comment (betsybeth) [inline hc/api/migrations/0027_auto_20180328_1452.py:9]: Add docstring and comments

**Comment by betsybeth** [comment]
Good implementation

File: hc/api/migrations/0027_auto_20180328_1452.py (hunk 5 above)
Comment (lakivisi-zz) [reply hc/api/migrations/0027_auto_20180328_1452.py:9]: 👎 because it is a migration file

**Pull request merged**